Libraries/Components/WebView/WebView.ios.js

```diff
@@ -109,301 +149,263 @@ const defaultRenderError = (errorDomain, errorCode, errorDesc) => (
  * history and configure various properties for the web content.
  */
-class WebView extends React.Component {
-  static JSNavigationScheme = JSNavigationScheme;
-  static NavigationType = NavigationType;
-  static propTypes = {
-    ...DeprecatedViewPropTypes,
 
-    html: deprecatedPropType(
-      PropTypes.string,
-      'Use the `source` prop instead.',
-    ),
+type Props = $ReadOnly<{|
+  ...ViewProps,
+  html?: string,
+  url?: string,
+  /**
+   * Loads static html or a uri (with optional headers) in the WebView.
+   */
+  source: SourceWithUri | SourceWithHtml | SourceNumber,
+  /**
+   * If true, use WKWebView instead of UIWebView.
+   * @platform ios
+   */
+  useWebKit?: boolean,
 
-    url: deprecatedPropType(PropTypes.string, 'Use the `source` prop instead.'),
+  /**
+   * Function that returns a view to show if there's an error.
+   */
+  renderError?: Function, // view to show if there's an error
+  /**
+   * Function that returns a loading indicator.
+   */
+  renderLoading?: Function,
+  /**
+   * Function that is invoked when the `WebView` has finished loading.
+   */
+  onLoad?: Function,
+  /**
+   * Function that is invoked when the `WebView` load succeeds or fails.
+   */
+  onLoadEnd?: Function,
+  /**
+   * Function that is invoked when the `WebView` starts loading.
+   */
+  onLoadStart?: Function,
+  /**
+   * Function that is invoked when the `WebView` load fails.
+   */
+  onError?: Function,
+  /**
+   * Boolean value that determines whether the web view bounces
+   * when it reaches the edge of the content. The default value is `true`.
+   * @platform ios
+   */
+  bounces?: boolean,
+  /**
+   * A floating-point number that determines how quickly the scroll view
+   * decelerates after the user lifts their finger. You may also use the
+   * string shortcuts `"normal"` and `"fast"` which match the underlying iOS
+   * settings for `UIScrollViewDecelerationRateNormal` and
+   * `UIScrollViewDecelerationRateFast` respectively:
+   *
+   *   - normal: 0.998
+   *   - fast: 0.99 (the default for iOS web view)
+   * @platform ios
+   */
+  decelerationRate: number | 'normal' | 'fast',
+  /**
+   * Boolean value that determines whether scrolling is enabled in the
+   * `WebView`. The default value is `true`.
+   * @platform ios
+   */
+  scrollEnabled?: boolean,
+  /**
+   * Controls whether to adjust the content inset for web views that are
+   * placed behind a navigation bar, tab bar, or toolbar. The default value
+   * is `true`.
+   */
+  automaticallyAdjustContentInsets?: boolean,
+  /**
+   * The amount by which the web view content is inset from the edges of
+   * the scroll view. Defaults to {top: 0, left: 0, bottom: 0, right: 0}.
+   * @platform ios
+   */
+  contentInset?: {
+    top: number,
+    left: number,
+    bottom: number,
+    right: number,
+  },
+  /**
+   * Function that is invoked when the `WebView` loading starts or ends.
+   */
+  onNavigationStateChange?: Function,
```

Comment:
We should strengthen this type. What I did earlier was look at how the prop was invoked in the component, and then follow the logic up to RCTWKWebView.m. Please let me know if you need help.

Reply:
I've solved it by myself!